### ISSUE TITLE
Don´t hard code the audio and video stream...

### ISSUE THREAD

**Issue by alex6dj** [tomaszzmuda/Xabe.FFmpeg#23]
Possibility to select the audio to use of those that exist in the original file, mainly for mkv.

**Comment by tomaszzmuda** [comment]
This issue will take some time to be done.
I have to think about implementation but i'm sure it will change API so you will have to wait some time for it.

**Comment by alex6dj** [comment]
Okay, I know you did not think it that way and the changes are going to be many. By the way excellent work with this project. One of the things I'm anxiously waiting for is implementing a conversion queue ...

**Comment by tomaszzmuda** [comment]
[Conversion queue](https://github.com/tomaszzmuda/Xabe.FFmpeg/wiki/Queue) look at this.
And thank you. It's meaning a lot for me :)

**Comment by alex6dj** [comment]
Okay, I was already checking the conversion queue. I know it's a preliminary version, but as a recommendation try to go a little further. For example this could be done with a Parallel.Foreach without any complication.

First the example has an error, you declare the same variable 2 times and you use it only once. For me it is good as a basic example but maybe someone else does not understand it well and does not know wy it gives an error ...

I currently use this

https://github.com/huysentruitw/process-manager

In my personal case what I would need in a conversion queue (some things are not implemented in the one I use) ...

- Possibility to start the queue and go adding jobs as needed.
- Completely stop or pause the queue (this may be not processing more jobs until it restarts).
- Maybe change the order of conversion or at least be able to give priority to certain works over others.
- Reporting the progress of each conversion is already done.
- Number of jobs in parallel (I see well that the number matches the processors count).
- Change the priority of processes (ProcessPriorityClass.BelowNormal) to approve the processor to the maximum when not using the machine but not to disturb when using it (this would be an idea for each process).

If there is anything else I let you know ...

By the way my native language is Spanish. I read English well but, so I helped the translator google, in case there is something you do not understand ...

regards

**Comment by tomaszzmuda** [comment]
Thank you for listening what you need.
I'm going to do some of this today.
I would be glad if we could talk. Can you write to me at [discord](https://discord.gg/aYDMN2E)?

**Comment by alex6dj** [comment]
Sorry sometimes we are in the wrong part of the earth. I am Cuban so the internet is limited just like the pages I have access to. We have to try another solution. I do not complain but sometimes makes communication difficult.

**Comment by tomaszzmuda** [comment]
So give me way to contant you in private message :)

**Comment by alex6dj** [comment]
Ok already have an acount in discordapp.com, it´s new to me. So let´s speak.
